mage_ai/cache/block.py

```diff
@@ -62,10 +71,13 @@ def build_key(self, block: Union[Dict], repo_path: str = None) -> str:
             configuration = block.configuration or {}
             file_path = block.file_path
 
-        if configuration:
-            file_source = (configuration or {}).get('file_source') or {}
-            if file_source and (file_source or {}).get('path'):
+        file_source = (configuration or {}).get('file_source') or {}
+        if file_source:
+            if (file_source or {}).get('path'):
                 return remove_base_repo_path_or_name((file_source or {}).get('path'))
```

Comment:
this looks a bit messy and confusing, i think you can just do this:

```suggestion
        if file_source.get('path'):
            return remove_base_repo_path_or_name(file_source.get('path'))
```